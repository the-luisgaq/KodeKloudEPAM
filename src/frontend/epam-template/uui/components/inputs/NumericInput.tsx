import { withMods, Overwrite } from '@epam/uui-core';
import { NumericInput as uuiNumericInput, NumericInputProps as uuiNumericInputProps } from '@epam/uui-components';
import { EditMode, IHasEditMode } from '../types';
import { settings } from '../../settings';

import textInputCss from './TextInput.module.scss';
import css from './NumericInput.module.scss';

const DEFAULT_MODE = EditMode.FORM;

type NumericInputMods = IHasEditMode & {
    /**
     * Defines component size
     * @default '36'
     */
    size?: '24' | '30' | '36' | '42' | '48';
};

export interface NumericInputModsOverride {}

function applyNumericInputMods(mods: NumericInputMods) {
    return [
        textInputCss.root,
        css.root,
        `uui-size-${mods.size || settings.numericInput.sizes.default}`,
        'uui-control-mode-' + (mods.mode || DEFAULT_MODE),
    ];
}

/** Represents the properties of a NumericInput component. */
export interface NumericInputProps extends uuiNumericInputProps, Overwrite<NumericInputMods, NumericInputModsOverride> {}

export const NumericInput = withMods<uuiNumericInputProps, NumericInputProps>(
    uuiNumericInput,
    applyNumericInputMods,
    (props) => {
        return {
            upIcon: settings.numericInput.icons.arrowIcon,
            downIcon: settings.numericInput.icons.arrowIcon,
            align: props.align ?? (props.mode === 'cell' ? 'right' : 'left'),
            disableArrows: props.disableArrows ?? props.mode === 'cell',
        };
    },
);
