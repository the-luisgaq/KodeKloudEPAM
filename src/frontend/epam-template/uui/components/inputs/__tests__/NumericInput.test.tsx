import React from 'react';
import { NumericInput } from '../NumericInput';
import { renderSnapshotWithContextAsync } from '@epam/uui-test-utils';

describe('NumericInput', () => {
    describe('snapshots', () => {
        it('should be rendered with minimum props', async () => {
            const tree = await renderSnapshotWithContextAsync(<NumericInput value={ null } onValueChange={ jest.fn } min={ 0 } max={ 50 } />);
            expect(tree).toMatchSnapshot();
        });

        it('should be rendered with maximum props', async () => {
            const tree = await renderSnapshotWithContextAsync(<NumericInput value={ null } onValueChange={ jest.fn } min={ 0 } max={ 50 } size="36" mode="inline" />);
            expect(tree).toMatchSnapshot();
        });
    });
});
