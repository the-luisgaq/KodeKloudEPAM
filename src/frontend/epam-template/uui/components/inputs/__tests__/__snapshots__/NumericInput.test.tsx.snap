// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumericInput snapshots should be rendered with maximum props 1`] = `
<DocumentFragment>
  <div
    class="container uui-input-box -clickable root root uui-size-36 uui-control-mode-inline"
    tabindex="-1"
  >
    <input
      class="uui-input align-left"
      inputmode="numeric"
      max="50"
      min="0"
      placeholder="0"
      tabindex="0"
      type="number"
      value=""
    />
    <div
      class="uui-numeric-input-button-group"
    >
      <div
        class="container uui-icon uui-enabled -clickable uui-numeric-input-up-button"
        role="button"
      >
        <svg
          class=""
        />
      </div>
      <div
        class="container uui-icon uui-enabled -clickable uui-numeric-input-down-button"
        role="button"
      >
        <svg
          class=""
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NumericInput snapshots should be rendered with minimum props 1`] = `
<DocumentFragment>
  <div
    class="container uui-input-box -clickable root root uui-size-36 uui-control-mode-form"
    tabindex="-1"
  >
    <input
      class="uui-input align-left"
      inputmode="numeric"
      max="50"
      min="0"
      placeholder="0"
      tabindex="0"
      type="number"
      value=""
    />
    <div
      class="uui-numeric-input-button-group"
    >
      <div
        class="container uui-icon uui-enabled -clickable uui-numeric-input-up-button"
        role="button"
      >
        <svg
          class=""
        />
      </div>
      <div
        class="container uui-icon uui-enabled -clickable uui-numeric-input-down-button"
        role="button"
      >
        <svg
          class=""
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
